### ISSUE TITLE
Compilation issue with gnucobol

### ISSUE THREAD

**Issue by braoult** [GaloisGirl/Coding#1]
As a challenge for myself, I am currently doing the AOC-2020 in both C and bash, and wanted to give Cobol a try. As a starting point, I had a look on your source to understand the basics to read a file (I totally forgot the basics). I got your example for day1/exercise 2, and tried to compile (gnucobol version 2.2.0).
```
$ cobc -x ex2-galoisgirl-cob.cob
ex2-galoisgirl-cob.cob: 2: error: invalid PROGRAM-ID 'Advent-of-Code-2020-day-1-part-2' - length is < 1 or > 31
```
The program-id seems too long, so I changed it to "AoC-2020-day-1-part-2". I was able to compile it, I did put input data in a "d1.input" file, but the result was wrong:
```
$ ./ex2-galoisgirl-cob 2>&1
011216496250
```
After adding some output, I noticed that the way you read the numbers into a 9(4) array may be a problem, especially the line:
`COMPUTE WS-ARRAY(I) = INPUTRECORD`. Apparently when numbers are less than 1,000, they are parsed wrongly. For instance, the number '123' would be parsed as '1230' into WS-ARRAY.
My question is: If you use gnucobol (cobc), is there some compilation option to make this line work ? Or did you change the input file, for example by left padding with zero(s) ?

Meantime, I made a new version, replacing the above line with `UNSTRING INPUTRECORD INTO WS-ARRAY(I)`. The result was then correct.

If you are interested, here is my own version: [https://github.com/braoult/Coding/blob/master/AdventOfCode2020/d01b.cob](https://github.com/braoult/Coding/blob/master/AdventOfCode2020/d01b.cob) .
Main changes: changed input file to stdin, and implemented a simple sort on array (bubble sort), to speed-up the main loop.

**Comment by GaloisGirl** [comment]
Happy to see someone is trying these!

> My question is: If you use gnucobol (cobc), is there some compilation option to make this line work ? Or did you change the input file, for example by left padding with zero(s) ?

I did pad the input on day 1 so that every line has 4 digits - I should add a comment about that. I dealt with variable length input since day 2 (`RECORD IS VARYING IN SIZE` etc.), and by day 9 I'm dealing with variable length numbers properly - I'm guessing the magic is done in `MOVE INPUTRECORD TO WS-NUMBER` with `INPUTRECORD PIC X(16)` and `WS-NUMBER PIC 9(16)`. Maybe `MOVE` accounts for the spaces and `COMPUTE` does not? I'm not sure.

> The program-id seems too long 

It's possible I added the "-part-1" as I was moving to part 2, so I didn't run it anymore.

Thanks for the remarks, I'll try and commit fixes when I have a minute.

**Comment by braoult** [comment]
Thanks,

As I am very new to Cobol (I learned it a little at school 40 years ago), I am far to be fluent in this language.
But I was really happy to make a 1st Cobol program for so long time !

Other topic: If you have interest in both Bash and C solutions for AoC, I just created a repo for my own implementations for days 1-12 : [https://github.com/braoult/adventofcode-2020](https://github.com/braoult/adventofcode-2020)

**Comment by GaloisGirl** [comment]
Fixed in 025decbc609fbd9c7dacbf8d058d86553c3b4e60

**Comment by GaloisGirl** [comment]
Fixed in 025decbc609fbd9c7dacbf8d058d86553c3b4e60
